xclim/indices/generic.py

```diff
@@ -424,41 +526,90 @@ def spell_length_statistics(
+
+    See Also
+    --------
+    spell_mask : The lower level functions that finds spells.
+    bivariate_spell_length_statistics : The bivariate version of this function.
     """
-    thresh = convert_units_to(
-        threshold,
-        data,
-        context=infer_context(standard_name=data.attrs.get("standard_name")),
+    thresh = convert_units_to(threshold, data, context="infer")
+    is_in_spell = spell_mask(data, window, win_reducer, op, thresh).astype(np.float32)
+    is_in_spell = select_time(is_in_spell, **indexer)
+
+    out = rl.resample_and_rl(
+        is_in_spell,
+        resample_before_rl,
+        rl.rle_statistics,
+        reducer=spell_reducer,
+        # The code above already ensured only spell of the minimum length are selected
+        window=1,
+        freq=freq,
     )
 
-    if window == 1:  # Fast path
-        is_in_spell = compare(data, op, thresh)
-    elif (win_reducer == "min" and op in [">", ">=", "ge", "gt"]) or (
-        win_reducer == "max" and op in ["`<", "<=", "le", "lt"]
-    ):
-        # Fast path for specific cases, this yields a smaller dask graph (rolling twice is expensive!)
-        # For these two cases, a day can't be part of a spell if it doesn't respect the condition itself
-        mask = compare(data, op, thresh)
-        # We need to filter out the spells shorter than "window"
-        # find sequences of consecutive respected constraints
-        cs_s = rl._cumsum_reset_on_zero(mask)
-        # end of these sequences
-        cs_s = cs_s.where(mask.shift({"time": -1}, fill_value=0) == 0)
-        # propagate these end of sequences
-        # the `.where(mask>0, 0)` acts a stopper
-        is_in_spell = cs_s.where(cs_s >= window).where(mask > 0, 0).bfill("time") > 0
-    else:
-        data_pad = data.pad(time=(0, window))
-        # The spell-wise value to test
-        # For example "win_reducer='sum'", we want the sum over the minimum spell length (window) to be above the thresh
-        spell_value = getattr(data_pad.rolling(time=window), win_reducer)()
-        # True at the end of a spell respecting the condition
-        mask = compare(spell_value, op, thresh)
-        # True for all days part of a spell that respected the condition (shift because of the two rollings)
-        is_in_spell = (mask.rolling(time=window).sum() >= 1).shift(time=-(window - 1))
-        # Cut back to the original size
-        is_in_spell = is_in_spell.isel(time=slice(0, data.time.size)).astype(float)
+    if spell_reducer == "count":
+        return out.assign_attrs(units="")
+    # All other cases are statistics of the number of timesteps
+    return to_agg_units(out, data, "count")
 
+
+@declare_relative_units(threshold1="<data1>", threshold2="<data2>")
+def bivariate_spell_length_statistics(
+    data1: xarray.DataArray,
+    threshold1: Quantified,
+    data2: xarray.DataArray,
+    threshold2: Quantified,
+    window: int,
+    win_reducer: str,
+    op: str,
+    spell_reducer: str,
```

Comment:
I think it's fine like this, maybe we don't want too much complexity, but let me point out a generalization that comes to mind: Having `op1` and `op2`.  For instance, `tasmax>thresh1`, `pr<thresh2`, i.e. `hot_and_dry_spells`.  See `_bivariate_spell_length_distribution` for instance.